src/spatialdata/_core/spatialdata.py

```diff
@@ -184,13 +229,167 @@ def from_elements_dict(elements_dict: dict[str, SpatialElement | AnnData]) -> Sp
+        instance_key: None | str = None,
+    ) -> None:
+        """Change the annotation target of a table currently having annotation metadata already.
+
+        Parameters
+        ----------
+        table
+            The table already annotating a SpatialElement.
+        region
+            The name of the target SpatialElement for which the table annotation will be changed.
+        region_key
+            The name of the region key column in the table. If not provided, it will be extracted from the table's uns
+            attribute.
+        instance_key
+            The name of the instance key column in the table. If not provided, it will be extracted from the table's uns
+            attribute.
+
+        Raises
+        ------
+        ValueError
+            If no region_key is provided, and it is not present in both table.uns['spatialdata_attrs'] and table.obs.
+        ValueError
+            If provided region_key is not present in table.obs.
+        """
+        attrs = table.uns[TableModel.ATTRS_KEY]
+        table_region_key = attrs.get(TableModel.REGION_KEY_KEY)
+        if not region_key:
+            if not table_region_key:
+                raise ValueError(
+                    "No region_key in table.uns and no region_key provided as argument. Please specify " "region_key."
+                )
+            if table.obs.get(attrs[TableModel.REGION_KEY_KEY]) is None:
+                raise ValueError(
+                    f"Specified region_key in table.uns '{table_region_key}' is not "
+                    f"present as column in table.obs. Please specify region_key."
+                )
+            TableModel()._validate_set_instance_key(table, instance_key)
+            check_target_region_column_symmetry(table, table_region_key, region)
+            attrs[TableModel.REGION_KEY] = region
+        else:
+            if region_key not in table.obs:
+                raise ValueError(f"'{region_key}' column not present in table.obs")
+
+            TableModel()._validate_set_instance_key(table, instance_key)
+            check_target_region_column_symmetry(table, table_region_key, region)
+            attrs[TableModel.REGION_KEY] = region
+
+    def set_table_annotates_spatialelement(
+        self,
+        table_name: str,
+        region: str | pd.Series,
+        region_key: None | str = None,
+        instance_key: None | str = None,
+    ) -> None:
+        """
+        Set the SpatialElement annotation target of a given AnnData table.
+
+        Parameters
+        ----------
+        table_name
+            The name of the table to set the annotation target for.
+        region
+            The name of the target element for the annotation. This can either be a string or a pandas Series object.
+        region_key
+            The region key for the annotation. If not specified, defaults to None.
+        instance_key
+            The instance key for the annotation. If not specified, defaults to None.
```

Comment:
I'd expain that when the table already contains these values they are reused (this is the case in which _change_table_annotation_target() is called)

Reply:
Resolved with [4960b35](https://github.com/scverse/spatialdata/pull/410/commits/4960b3523c1f27d8db607e5295e57117070c50e1).